docs/develop/create-sso-office-add-ins-nodejs.md

```diff
@@ -1,14 +1,574 @@
+    - The code minimizes the data that must come from Microsoft Graph by specifying only the property we need ("name") and only the top 10 folder or file names.
+
+    ```javascript
+    const graphToken = req.get('access_token');    
+    const graphData = await getGraphData(graphToken, "/me/drive/root/children", "?$select=name&$top=10");
+    ```
+
+1. Replace `TODO 16` with the following. About this code, note:
+
+    - If Microsoft Graph returns an error, such as invalid or expired token, there will be a code property in the returned object set to a HTTP status (e.g. 401). The code relays the error to the client. It will br caught in the `.fail` callback of `makeGraphApiCall`.
+    - Microsoft Graph data includes OData metadata and eTags that the add-in does not need, so the code constructs a new array containing only the file names to send to the client.
+
+    ```javascript
+    if (graphData.code) {
+        next(createError(graphData.code, "Microsoft Graph error " + JSON.stringify(graphData)));
+    }
+    else 
+    {
+        const itemNames = [];
+        const oneDriveItems = graphData['value'];
+        for (let item of oneDriveItems) {
+            itemNames.push(item['name']);
+        }
+
+        res.send(itemNames)
+    }
+    ```
+
+1. Save and close the file.
+
+## Run the project
+
+1. Ensure that you have some files in your OneDrive so that you can verify the results.
+
+1. Open a command prompt in the root of the root of the `\Complete` folder. 
+
+1. Run the command `npm start`. 
+
+1. You need to sideload the add-in into an Office application (Excel, Word, or PowerPoint) to test it. The instructions depend on your platform. There are links to instructions at [Sideload an Office Add-in for Testing](https://docs.microsoft.com/office/dev/add-ins/testing/test-debug-office-add-ins#sideload-an-office-add-in-for-testing).
+
+1. In the Office application, on the **Home** ribbon, select the **Show Add-in** button in the **SSO Node.js** group to open the task pane add-in.
+
+1. Click the **Get OneDrive File Names** button. If you are logged into Office with either a Work or School (Office 365) account or Microsoft Account, and SSO is working as expected, the first 10 file and folder names in your OneDrive for Business are inserted into the document. (It may take as much as 15 seconds the first time.) If you are not logged in, or you are in a scenario that does not support SSO, or SSO is not working for any reason, you will be prompted to log in. After you log in, the file and folder names appear.
+
+> [!NOTE]
+> If you were previously signed on to Office with a different ID, and some Office applications that were open at the time are still open, Office may not reliably change your ID even if it appears to have done so. If this happens, the call to Microsoft Graph may fail or data from the previous ID may be returned. To prevent this, be sure to *close all other Office applications* before you press **Get OneDrive File Names**.
```

Comment:
```suggestion
> If you were previously signed into Office with a different ID, and some Office applications that were open at the time are still open, Office may not reliably change your ID even if it appears to have done so. If this happens, the call to Microsoft Graph may fail or data from the previous ID may be returned. To prevent this, be sure to *close all other Office applications* before you press **Get OneDrive File Names**.
```